benchmark/src/bench_client.rs

```diff
@@ -91,16 +104,101 @@ impl BenchClient {
         &mut self,
         request: PutRequest,
     ) -> Result<PutResponse, BenchClientError> {
-        if self.use_curp_client {
-            let response = self.xline_client.kv_client().put(request).await?;
-            Ok(response)
-        } else {
-            let opts = from_request_to_option(&request);
-            let response = self
-                .etcd_client
-                .put(request.key(), request.value(), Some(opts))
-                .await?;
-            Ok(response.into())
+        match self.kv_client {
+            KVClient::Xline(ref mut xline_client) => {
+                let response = xline_client.kv_client().put(request).await?;
+                Ok(response)
+            }
+            KVClient::Etcd(ref mut etcd_client) => {
+                let opts = from_request_to_option(&request);
+                let response = etcd_client
+                    .put(request.key(), request.value(), Some(opts))
+                    .await?;
+                Ok(response.into())
+            }
+        }
+    }
+
+    /// Send `RangeRequest` by `XlineClient` or `EtcdClient`
+    ///
+    /// # Errors
+    ///
+    /// If `XlineClient` or `EtcdClient` failed to send request
+    #[inline]
+    #[allow(dead_code)]
+    pub(crate) async fn get(
+        &mut self,
+        request: RangeRequest,
+    ) -> Result<RangeResponse, BenchClientError> {
+        match self.kv_client {
+            KVClient::Xline(ref mut xline_client) => {
+                let response = xline_client.kv_client().range(request).await?;
+                Ok(response)
+            }
+            KVClient::Etcd(ref mut etcd_client) => {
+                let response = etcd_client.get(request.key(), None).await?;
+                Ok(response.into())
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+#[allow(clippy::panic)]
+#[allow(clippy::unwrap_used)]
+#[allow(clippy::indexing_slicing)]
+mod test {
+    use crate::bench_client::{BenchClient, ClientOptions, PutRequest};
+    use xline_client::types::kv::RangeRequest;
+    use xline_test_utils::Cluster;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_new_xline_client() {
+        // create xline client
+        let mut cluster = Cluster::new(3).await;
+        cluster.start().await;
+        let use_curp_client = true;
+        let config = ClientOptions::default();
+        let mut client = BenchClient::new(cluster.addrs(), use_curp_client, config)
+            .await
+            .unwrap();
+        //check xline client put value exist
+        let request = PutRequest::new("put", "123");
+        let _put_response = client.put(request).await;
+        let range_request = RangeRequest::new("put");
+        match client.get(range_request).await {
```

Comment:
You can directly unwrap the result if you do believe that it's not an error. 
```suggestion
        let response = client.get(range_request).await.unwrap();
        assert_eq!(response.kvs[0].value, b"123");
```